src/blocks/agc.rs

```diff
@@ -0,0 +1,257 @@
+use num_complex::ComplexFloat;
+use futuresdr_pmt::Pmt;
+use futures::FutureExt;
+
+use crate::anyhow::Result;
+use crate::runtime::Block;
+use crate::runtime::BlockMeta;
+use crate::runtime::BlockMetaBuilder;
+use crate::runtime::Kernel;
+use crate::runtime::MessageIo;
+use crate::runtime::MessageIoBuilder;
+use crate::runtime::StreamIo;
+use crate::runtime::StreamIoBuilder;
+use crate::runtime::WorkIo;
+
+
+pub struct AGC<T> {
+    // Minimum value that has to be reached in order for AGC to start adjusting gain.
+    squelch: T,
```

Comment:
Why use type T for members? I think it would make more sense to just use a fixed type, like f32. All these gain related values should be real, so if I try to use a complex sample type it would be weird. If it is desired to use different real gain types (f32, f64) then an additional type could be used to differentiate it from the sample type. I'd just stick with f32 for now.